src/main/php/com/amazon/aws/ServiceEndpoint.class.php

```diff
@@ -206,7 +211,7 @@ public function request(string $method, string $target, array $headers= [], $pay
       $method,
       $target,
       $params,
-      hash(SignatureV4::HASH, $payload ?? ''),
+      $hash ?? $headers['x-amz-content-sha256'] ?? self::$NO_CONTENT,
```

Comment:
Can't we *always* just send the `x-amz-content-sha256` header?

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html states:

> The `x-amz-content-sha256` header is required for all AWS Signature Version 4 requests. It provides a hash of the request payload. If there is no payload, you must provide the hash of an empty string.